### PR TITLE
feat(Row): gutter support vertical space

### LINKED ISSUE #12431: [Feature Request] gutter  支持垂直间距
### 这个功能解决了什么问题？

循环出来的col, 自行换行后, 第二行的都紧挨着上方, 需要一个方便的方法设置行间距

### 你期望的 API 是什么样子的？

```vue
<van-row :gutter="[10, 10]">
  <van-col v-for="item in list" :key="item" :span="8">1</van-col>
</van-row>
```

### PR DESCRIPTION
**Pull request by cc-hearts**
The `gutter` property of the `Row` component supports setting both horizontal and vertical spacing as an array.

close #12431

### CODE REVIEW

**Comment by inottn** [comment]
Does row-gap have compatibility issues on mobile devices?

**Comment by cc-hearts** [comment]
image

all major browsers seem to support it. Does it need to be downgraded to the way margin is calculated?

**Comment by inottn** [comment]
> image
> all major browsers seem to support it. Does it need to be downgraded to the way margin is calculated?

The compatibility of row-gap in grid and flex layouts is different.

**Comment by cc-hearts** [comment]
The flex layout compatibility is a bit bad, I'll try to refactor it.

File: packages/vant/src/col/README.md [hunk 1]
```diff
@@ -45,12 +45,22 @@ Layout are based on 24-column. The attribute `span` in `Col` means the number of
 
 Set grid spacing using `gutter` attribute. The default value is 0.
 
+If you want to set the vertical spacing, you can set `[horizontal,vertical]` as an array.
```
Comment (inottn) [inline packages/vant/src/col/README.md:48]:
```suggestion
If you want to set the vertical spacing, you can set `[horizontal, vertical]` as an array.
```

File: packages/vant/src/row/Row.tsx [hunk 2]
```diff
@@ -94,7 +101,26 @@ export default defineComponent({
...
+      const spaces: VerticalSpaces = [];
+      if (Array.isArray(gutter) && gutter.length > 1) {
+        const bottom = Number(gutter[1]) || 0;
+        if (bottom <= 0) {
+          return spaces;
+        }
+        groups.value.forEach((group, index) => {
+          group.forEach(() => {
+            if (index !== groups.value.length - 1) {
+              spaces.push({ bottom: bottom });
```
Comment (inottn) [inline packages/vant/src/row/Row.tsx:115]:
```suggestion
              spaces.push({ bottom });
```

**Comment by cc-hearts** [comment]
Do you have time for a review?

**Comment by inottn** [comment]
In general, PR will be handled collectively on the weekend~

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
thank you for your contribution!

**Pull request merged**